storage/onpremise/mongodb/secrets.tf

```diff
@@ -1,14 +1,26 @@
+data "kubernetes_secret" "mongodb_certificates" {
+  metadata {
+    name      = "${helm_release.mongodb.name}-ca"
+    namespace = var.namespace
+  }
+  # Kubernetes provider syntax for retrieving base64 secret data
+  binary_data = {
```

Comment:
As certificates and keys are encoded in PEM (plain ASCII), I think you can use `data` instead of `binary_data` and avoid base64 decoding in terraform.